src/shogun/machine/gp/KLDualInferenceMethod.cpp

```diff
@@ -507,14 +534,20 @@ void CKLDualInferenceMethod::update_alpha()
 
 float64_t CKLDualInferenceMethod::optimization()
 {
-	KLDualInferenceMethodMinimizer *minimizer=dynamic_cast<KLDualInferenceMethodMinimizer*>(m_minimizer);
-	REQUIRE(minimizer,"The minimizer must be an instance of KLDualInferenceMethodMinimizer\n"); 
+	CKLDualInferenceMethodMinimizer *minimizer=dynamic_cast<CKLDualInferenceMethodMinimizer*>(m_minimizer);
+	REQUIRE(minimizer,"The minimizer must be an instance of KLDualInferenceMethoMinimizer\n"); 
         KLDualInferenceMethodCostFunction* cost_fun=new  KLDualInferenceMethodCostFunction();
 	cost_fun->set_target(this);
+	bool cleanup=false;
+
+	if(this->ref_count()>1)
```

Comment:
Mmmh I see,
Not sure we would do that in  here. Where is the circular thing biting you?